src/core_read.cpp

```diff
@@ -59,6 +60,14 @@ CScript ParseScript(const std::string& s)
         {
             // Number
             int64_t n = atoi64(*w);
+
+            //limit the range of numbers ParseScript accepts in decimal
+            //since numbers outside -0xFFFFFFFF...0xFFFFFFFF are illegal in scripts
+            if(n > static_cast<int64_t>(0xffffffff) || n < -1*static_cast<int64_t>(0xffffffff)) {
```

Comment:
Any reason this needs a static cast?
 
```suggestion
            if (n > 0xffffffff || n < -0xffffffff) {
```

Reply:
The second static_cast is strictly necessary, see this example : https://godbolt.org/z/Ko33NG

The first one is supperficial (implicit), I just thought that the symetry looked better.

Reply:
Explicit is better than implicit :)

Reply:
I think with C++11 you can also get a safer way to achieve the same: `int64_t{bla}`. If the value doesn't fit in the integer type, it won't compile. Whereas with static_cast it would compile.

Reply:
Thanks, updated the branch accordingly,